### PR TITLE
Remove compound key (‘key’,’scope’) before removing scope column

### PR DESCRIPTION
**Pull request by assmith**
I found that the compound index previously on ('key','scope') prevented the scope column from being removed during this migration.  This PR suggests removing the key then deleting the column then recreating the key as per the current schema in 20180220023243_create_settings_table.

As per the current schema I've left the index name as _idx_scope_name_, although maybe it should just be called _idx_name_ now?

### CODE REVIEW

**Comment by rijkvanzanten** [comment]
Thanks @assmith!

I think you're right. We should probably call the new index just idx_key instead of having the word scope in there (to prevent future confusion). Also, let's call it idx_key instead of idx_name 🙂

**Comment by assmith** [comment]
Sounds good, I wasn't sure if there was an internal reason for using _name_, _key_ would be better though.

File: migrations/upgrades/schemas/20181123171520_remove_scope.php [hunk 1]
```diff
@@ -8,7 +8,12 @@ class RemoveScope extends AbstractMigration
     public function up()
     {
         $table = $this->table('directus_settings');
+        $table->removeIndexByName('idx_scope_name');
         $table->removeColumn('scope')
               ->save();
+        $table->addIndex(['key'], [
+            'unique' => true,
+            'name' => 'idx_scope_name'
```
Comment (WellingGuzman) [inline migrations/upgrades/schemas/20181123171520_remove_scope.php]: We can change this to `idx_key

**Comment by WellingGuzman** [comment]
Thanks @assmith!

**Pull request merged**